modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -124,6 +124,15 @@
 ;; See https://docs.microsoft.com/en-us/sql/t-sql/functions/date-and-time-data-types-and-functions-transact-sql for
 ;; details on the functions we're using.
 
+(defn- date-trunc
+  "Truncates `expr` to the specified `unit`."
+  [unit expr]
+  (hsql/call :dateadd (hsql/raw (name unit)) (hx/- 1 (date-part unit expr)) expr))
```

Comment:
```suggestion
  (hsql/call :dateadd (hx/literal unit) (hx/- 1 (date-part unit expr)) expr))
```

does this works?

Reply:
Nope. `(hx/literal unit)` wraps unit in single quotes.